lib/replica_pools/query_cache.rb

```diff
@@ -32,7 +34,7 @@ def select_all(*args)
       if raw_binds.blank? && relation.is_a?(ActiveRecord::Relation)
         arel, binds = relation.arel, relation.bind_values
       else
-        arel, binds = relation, raw_binds
+        arel, binds = relation, Array(raw_binds)
```

Comment:
At first i was confused because of the conditional - `if raw_binds.blank?`. If raw_binds was `nil` `blank?` would be true. But of course there's the second part of the conditional `relations.is_a?(ActiveRecord::Relation`. 

Sorry - this is not really a comment for anyone but me (haha). Trying to reason through things. I don't really know what `relation` is when it's not an `ActiveRecord::Relation` but I don't think it matters if I know?

Reply:
Good q, I can't say for sure what are all the types that fall into this method but the specific case we have is `pluck` which is class `Arel::SelectManager` which is not an `ActiveRecord::Relation` so it falls into the latter case.